### PR TITLE
feature: Operator search component

### PR DESCRIPTION
**Pull request by RobCattonKPMG**
Added a new component to the site for searching for operators, which is used on the consequence operator page. Note that the same behaviour is expected via the API - only the first operator in the selected list will be used.

### CODE REVIEW

File: site/schemas/consequence.schema.ts [hunk 1]
```diff
@@ -110,3 +107,20 @@ export const consequenceSchema = z.discriminatedUnion("consequenceType", [
 ]);
 
 export type Consequence = z.infer<typeof consequenceSchema>;
+
+export const operatorSchema = z.object({
+    id: z.number(),
+    nocCode: z.string(),
+    operatorPublicName: z.string(),
+    vosaPsvLicenseName: z.string().optional(),
```
Comment (alainajaffer) [inline site/schemas/consequence.schema.ts:115]: will we need all of these attributes?

File: site/testData/mockData.ts [hunk 2]
```diff
@@ -420,3 +420,76 @@ export const ptSituationElementWithMultipleConsequences = {
...
+    },
+    {
+        id: 6,
+        nocCode: "8H",
+        operatorPublicName: "Highland Airways",
+        vosaPsvLicenseName: "Highland Airways",
+        opId: "136620",
+        pubNmId: "94334",
+        nocCdQual: "",
+        changeDate: "",
```
Comment (alainajaffer) [inline site/testData/mockData.ts:489]: Are all of these needed?

File: site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx [hunk 3]
```diff
@@ -76,16 +93,47 @@ const CreateConsequenceOperator = (props: CreateConsequenceOperatorProps): React
...
+                                        ],
+                                    };
+                                })}
+                            />
+                        ) : null}
+
+                        <input
+                            type="hidden"
+                            name="consequenceOperator"
+                            value={selectedOperators.length > 0 ? selectedOperators[0].nocCode : undefined}
```
Comment (alainajaffer) [inline site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx:136]: should the else here be null?

File: site/schemas/consequence.schema.ts (hunk 1 above)
Comment (RobCattonKPMG) [reply site/schemas/consequence.schema.ts:115]: Yeah I'm not sure - the API call will be changing early next week, so I figured I'd add it all and we can cut it back as needed.

File: site/testData/mockData.ts (hunk 2 above)
Comment (RobCattonKPMG) [reply site/testData/mockData.ts:489]: Its just copy and pasted data that came back from the API.

File: site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx (hunk 3 above)
Comment (RobCattonKPMG) [reply site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx:136]: Yeah I can change it to null.

Comment (RobCattonKPMG) [reply site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx:136]: Ah - null is not an allowed value.

File: site/schemas/consequence.schema.ts (hunk 1 above)
Comment (alainajaffer) [reply site/schemas/consequence.schema.ts:115]: ok that is fine for now

Comment (RobCattonKPMG) [reply site/schemas/consequence.schema.ts:115]: Its not about whats being displayed - we only need the NOC and the Operator Name (operatorPublicName) - its about what we need in the back end for when we're using the operator information.

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**